central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -287,17 +294,31 @@ func (s *serviceImpl) GetComplianceClusterStats(ctx context.Context, request *v2
 	}
 
 	// Lookup the integrations to get the status
+	clusterLastScan := make(map[string]*types.Timestamp, len(scanResults))
 	clusterErrors := make(map[string][]string, len(scanResults))
 	for _, result := range scanResults {
+		// Get the integrations if we can.  If we cannot, it could be an externally configured
+		// scan and thus we will not have a matching integration.
 		integrations, err := s.integrationDS.GetComplianceIntegrationByCluster(ctx, result.ClusterID)
-		if err != nil || len(integrations) != 1 {
-			return nil, errors.Errorf("Unable to retrieve cluster %q", result.ClusterID)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Unable to retrieve configuration for cluster %q", result.ClusterID)
 		}
-		clusterErrors[result.ClusterID] = integrations[0].GetStatusErrors()
+		if len(integrations) == 1 {
```

Comment:
We expect one compliance integration for each cluster in the happy path right?

Reply:
Integrations,  yes.  One installation of CO per cluster.

Reply:
Actually, I need to adjust the changes for the integration here we should report an error status if there are more than one integration.  When I made these updates I had configurations in mind (I'm moving too fast).